task/sast-snyk-check/0.4/sast-snyk-check.yaml

```diff
@@ -0,0 +1,280 @@
+        SNYK_EXIT_CODE=$?
+        set -e
+        test_not_skipped=0
+        SKIP_MSG="We found 0 supported files"
+        grep -q "$SKIP_MSG" stdout.txt || test_not_skipped=$?
+
+        if [[ "$SNYK_EXIT_CODE" -eq 0 ]] || [[ "$SNYK_EXIT_CODE" -eq 1 ]]; then
+          # In order to generate csdiff/v1, we need to add the whole path of the source code as Snyk only provides an URI to embed the context
+          (cd  "${SOURCE_CODE_DIR}" && csgrep --mode=json --embed-context=3 "${SOURCE_CODE_DIR}"/sast_snyk_check_out.json) \
+            | csgrep --mode=json --strip-path-prefix="source/"  \
+            > sast_snyk_check_out_all_findings.json
+
+          echo "Results:"
+          (set -x && csgrep --mode=evtstat sast_snyk_check_out_all_findings.json)
+
+          # We check if the KFP_GIT_URL variable is set to apply the filters or not
+          if [[ -z "${KFP_GIT_URL}" ]]; then
+            echo "KFP_GIT_URL variable not defined. False positives won't be filtered"
+            mv sast_snyk_check_out_all_findings.json filtered_sast_snyk_check_out.json
+          else
+            echo "Filtering false positives in results files using csfilter-kfp..."
+
+            CMD=(
+              csfilter-kfp
+              --verbose
+              --kfp-git-url="${KFP_GIT_URL}"
+              --project-nvr="${PROJECT_NAME}"
+            )
+
+            if [ "${RECORD_EXCLUDED}" == "true" ]; then
+              CMD+=(--record-excluded="excluded-findings.json")
+            fi
+
+            set +e
+            "${CMD[@]}" sast_snyk_check_out_all_findings.json > filtered_sast_snyk_check_out.json
+            status=$?
+            set -e
+            if [ "$status" -ne 0 ]; then
+              echo "Error: failed to filter known false positives" >&2
+              return 1
+            else
+              echo "Message: Succeed to filter known false positives" >&2
+            fi
+            echo "Results after filtering:"
+            (set -x && csgrep --mode=evtstat filtered_sast_snyk_check_out.json)
+          fi
+
+          csgrep --mode=sarif filtered_sast_snyk_check_out.json > no_stats_sast_snyk_check_out.sarif
+
+          # Recording of scan stats
+          echo "-------------"
+          echo "Scan stats"
+
+          total_files=$(jq '[.runs[0].properties.coverage[].files] | add' "${SOURCE_CODE_DIR}"/sast_snyk_check_out.json)
+          supported_files=$(jq '[.runs[0].properties.coverage[] | select(.type == "SUPPORTED") | .files] | add' "${SOURCE_CODE_DIR}"/sast_snyk_check_out.json)
+
+          # We make sure the values are not null (jq returns null for empty lists)
+          total_files=${total_files:-0}
+          supported_files=${supported_files:-0}
+
+          coverage_ratio=0
+          if (( total_files > 0 )); then
+              coverage_ratio=$((supported_files * 100 / total_files))
+          fi
+        
+          csgrep --mode=sarif --set-scan-prop snyk-scanned-files-coverage:${coverage_ratio} \
```

Comment:
I would not run `csgrep` twice.  The additional decoding and encoding of SARIF could be expensive on scans that produce too many findings. We can simply append the arguments to the original invocation of `csgrep`.

Reply:
I have modified the calls to `csgrep`and only one is done.